osbs/api.py

```diff
@@ -966,3 +967,31 @@ def get_resource_quota(self, quota_name):
     @osbsapi
     def can_orchestrate(self):
         return self.build_conf.get_can_orchestrate()
+
+    @osbsapi
+    def create_config_map(self, name, data):
+        """
+        Create an ConfigMap object
+
+        Raises exception on error
+
+        :param name: str, name of configMap
+        :param data: dict, dictionary of data to be stored
+        """
+        config_data_file = os.path.join(self.os_conf.get_build_json_store(), 'config_map.json')
+        config_data = json.load(open(config_data_file))
+        config_data['metadata']['name'] = name
+        config_data['metadata']['data'] = data
```

Comment:
It does matter though: OpenShift is on the other side of that PUT request, and it requires things to be in the correct places. It would be e.g. free to ignore keys it doesn't know.

The `graceful_chain_get` part is when we're parsing objects provided to us by OpenShift.